### PR TITLE
[4.0] Workflows checked out / checkin

### PR DESCRIPTION
**Pull request by chmst**
# Summary of Changes
Checked_out and Checkin now works in com_workflow in the same way as in all other components

### Testing Instructions
Open a workflow or stage or transition, close the browser window. Login again and see the locked icon in list views.
Klick on the locked icon to chckin the item. If you have permission for checkin, the item will be unlocked.

### CODE REVIEW

**Comment by richard67** [comment]
Please fix PHP code style. These errors are reported by Drone. Just click on link "[Details](https://ci.joomla.org/joomla/joomla-cms/25337)" right beside the "continuous-integration/drone/pr — Build is failing" label at the bottom of this PR in the "Checks" area.

**Comment by richard67** [comment]
Ah I‘ve forgotten: After following the link, select PHPCS in the list of tests shown at the left hand side to see the PHP code style log.

**Comment by chmst** [comment]
@richard67 can you help me? The mySql check fails. but I don't see the problem.

**Comment by richard67** [comment]
@chmst Failure of mysql system test seems to me to be unrelated to this PR. As far as I know they still fail sometimes due to checking things which are not visible yet, e.g. due to CSS transitions. But I am not good in reading those system test logs (the PCPHS is more clear). Maybe someone of the automated testing team can tell? @HLeithner .

File: administrator/components/com_workflow/Model/StagesModel.php [hunk 1]
```diff
@@ -159,6 +161,10 @@ public function getListQuery()
 			$query->where($db->quoteName('s.condition') . ' = ' . (int) $db->escape($condition));
 		}
 
+		// Join over the users for the checked out user.
+		$query->select('uc.name AS editor')
```
Comment (Quy) [inline administrator/components/com_workflow/Model/StagesModel.php]:
```suggestion
		$query->select($db->quoteName('uc.name', 'editor'))
```

File: administrator/components/com_workflow/Model/StagesModel.php [hunk 2]
```diff
@@ -159,6 +161,10 @@ public function getListQuery()
...
+			->join('LEFT', '#__users AS uc ON uc.id = s.checked_out');
```
Comment (Quy) [inline administrator/components/com_workflow/Model/StagesModel.php]:
```suggestion
			->join('LEFT', $db->quoteName('#__users', 'uc'), $db->quoteName('uc.id') . ' = ' . $db->quoteName('s.checked_out'));
```

**Comment by 1apweb** [comment]
I have tested this item :white_check_mark: successfully on efef56c824d91ebf47e1b39bfdc7114c83a61398<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

File: administrator/components/com_workflow/Model/StagesModel.php (hunk 1 above)
Comment (Quy) [reply administrator/components/com_workflow/Model/StagesModel.php]: Please make similar changes to the other 2 instances.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on c3db49d318a50cd444d3e4028ad4ebe3facdf8d7<br /><br />Tested with PostgreSQL 10.10 and error reporting = Development and watching database log file and PHP log.

Everything works as expected and there are no errors or warning.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by richard67** [comment]
In addition to the testing instructions, I have also tested that the locked out item is correctly shown on the "Global Check-In" page and the number of locked out items is shown correctly on the system dashboard, and that both checking in with lock icon and on the "Global Check-In" page works.

Null date is also ok.

**Comment by kishorchate** [comment]
I have tested this item :white_check_mark: successfully on c3db49d318a50cd444d3e4028ad4ebe3facdf8d7<br /><br />#pbf19 #test1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by ttplpoojak** [comment]
I have not tested this item.<br /><br />#pbf19 #test2

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by ttplpoojak** [comment]
I have tested this item :white_check_mark: successfully on c3db49d318a50cd444d3e4028ad4ebe3facdf8d7<br /><br />#pbf19 #test2

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by richard67** [comment]
@alikon I've just tried to set this PR to RTC but wass not successful. Could you do that?

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by richard67** [comment]
Thanks

File: administrator/components/com_workflow/Model/TransitionsModel.php [hunk 3]
```diff
@@ -158,6 +160,10 @@ public function getListQuery()
 			)
 			->leftJoin($joinTo);
 
+		// Join over the users for the checked out user.
+		$query->select('uc.name AS editor')
```
Comment (Quy) [inline administrator/components/com_workflow/Model/TransitionsModel.php]:
```suggestion
		$query->select($db->quoteName('uc.name', 'editor'))
```

File: administrator/components/com_workflow/Model/WorkflowsModel.php [hunk 4]
```diff
@@ -234,6 +236,10 @@ public function getListQuery()
 			$query->where('(' . $db->quoteName('w.title') . ' LIKE ' . $search . ' OR ' . $db->quoteName('w.description') . ' LIKE ' . $search . ')');
 		}
 
+		// Join over the users for the checked out user.
+		$query->select('uc.name AS editor')
```
Comment (Quy) [inline administrator/components/com_workflow/Model/WorkflowsModel.php]:
```suggestion
		$query->select($db->quoteName('uc.name', 'editor'))
```

File: administrator/components/com_workflow/Model/TransitionsModel.php [hunk 5]
```diff
@@ -158,6 +160,10 @@ public function getListQuery()
...
+			->join('LEFT', '#__users AS uc ON uc.id = t.checked_out');
```
Comment (Quy) [inline administrator/components/com_workflow/Model/TransitionsModel.php]:
```suggestion
			->join('LEFT', $db->quoteName('#__users', 'uc'), $db->quoteName('uc.id') . ' = ' . $db->quoteName('t.checked_out'));
```

File: administrator/components/com_workflow/Model/WorkflowsModel.php [hunk 6]
```diff
@@ -234,6 +236,10 @@ public function getListQuery()
...
+			->join('LEFT', '#__users AS uc ON uc.id = w.checked_out');
```
Comment (Quy) [inline administrator/components/com_workflow/Model/WorkflowsModel.php]:
```suggestion
			->join('LEFT', $db->quoteName('#__users', 'uc'), $db->quoteName('uc.id') . ' = ' . $db->quoteName('w.checked_out'));
```

**Comment by Quy** [comment]
@richard67 I set the status to Ready To Commit and write RTC in the Issue Tracker. I don't do it in GitHub. Give that a try.

**Comment by richard67** [comment]
@chmst Please apply the changes suggested by @Quy .

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 6427bedf68d7d53599b8e60dcfb451d1a6baadb6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Review by richard67 — APPROVED** [review]

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by richard67** [comment]
PR had enough good tests, changes after that were trivial (names quoting an 4.0 style join syntax for db access). I've tested that and reviewed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 6427bedf68d7d53599b8e60dcfb451d1a6baadb6<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26813">issues.joomla.org/tracker/joomla-cms/26813</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**